### PR TITLE
Add SetWindowKeepAspectRatio

### PR DESCRIPTION
**Pull request by tomlister**
This PR adds a new method that fixes the aspect ratio while resizing.

Tested on:
- [x] Ubuntu
- [x] Windows (soon, maybe)
- [ ] Darwin (I no longer use macOS)

### CODE REVIEW

**Comment by tomlister** [comment]
for some reason all these old commits are showing up :(

File: window.go [hunk 1]
```diff
@@ -220,6 +220,14 @@ func SetWindowSizeLimits(minw, minh, maxw, maxh int) {
 	}
 }
 
+// SetWindowKeepAspectRatio sets whether the window should keep its aspect ratio while resizing.
+
```
Comment (hajimehoshi) [inline window.go:224]: Remove this empty line.

**Review by hajimehoshi — COMMENTED** [review]
Seems good. Let's test the other platforms if possible!

**Comment by hajimehoshi** [comment]
I'll test macOS later.

**Comment by tomlister** [comment]
Cheers

**Comment by tomlister** [comment]
Works well on Windows.

**Comment by tomlister** [comment]
Hmm, I found a bug on windows. Nevermind.

**Comment by tomlister** [comment]
```
fatal error: sync: unlock of unlocked mutex

goroutine 92 [running]:
runtime.throw({0x41ca7e, 0x0})
        /usr/local/go/src/runtime/panic.go:1198 +0x76 fp=0xc000789de0 sp=0xc000789db0 pc=0x2b5bb6
sync.throw({0x41ca7e, 0x2b85c7})
        /usr/local/go/src/runtime/panic.go:1184 +0x1e fp=0xc000789e00 sp=0xc000789de0 pc=0x2dcb9e
sync.(*Mutex).unlockSlow(0x69f2c8, 0xffffffff)
        /usr/local/go/src/sync/mutex.go:196 +0x3c fp=0xc000789e28 sp=0xc000789e00 pc=0x2e901c
sync.(*Mutex).Unlock(0x2b8516)
        /usr/local/go/src/sync/mutex.go:190 +0x29 fp=0xc000789e48 sp=0xc000789e28 pc=0x2e8fa9
github.com/hajimehoshi/ebiten/v2/internal/atlas.BeginFrame·dwrap·5()
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/atlas/image.go:808 +0x26 fp=0xc000789e60 sp=0xc000789e48 pc=0x375006
github.com/hajimehoshi/ebiten/v2/internal/atlas.BeginFrame()
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/atlas/image.go:826 +0xf4 fp=0xc000789ed0 sp=0xc000789e60 pc=0x374ed4
github.com/hajimehoshi/ebiten/v2/internal/buffered.BeginFrame()
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/buffered/image.go:38 +0x17 fp=0xc000789ee0 sp=0xc000789ed0 pc=0x3759b7
github.com/hajimehoshi/ebiten/v2.(*uiContext).updateFrame(0x64a060, 0xc00050f020)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/uicontext.go:157 +0x72 fp=0xc000789f30 sp=0xc000789ee0 pc=0x3cda72
github.com/hajimehoshi/ebiten/v2.(*uiContext).ForceUpdateFrame(0xc000322a80)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/uicontext.go:147 +0x49 fp=0xc000789f50 sp=0xc000789f30 pc=0x3cd9c9
github.com/hajimehoshi/ebiten/v2/internal/uidriver/glfw.(*UserInterface).registerWindowSetSizeCallback.func1.1()
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/uidriver/glfw/ui.go:764 +0x138 fp=0xc000789fa0 sp=0xc000789f50 pc=0x3c3418
github.com/hajimehoshi/ebiten/v2/internal/uidriver/glfw.(*UserInterface).runOnAnotherThreadFromMainThread.func2()
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/uidriver/glfw/run_notsinglethread.go:78 +0x85 fp=0xc000789fe0 sp=0xc000789fa0 pc=0x3c0d05
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1581 +0x1 fp=0xc000789fe8 sp=0xc000789fe0 pc=0x2e1a61
created by github.com/hajimehoshi/ebiten/v2/internal/uidriver/glfw.(*UserInterface).runOnAnotherThreadFromMainThread
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/uidriver/glfw/run_notsinglethread.go:76 +0x21f

goroutine 1 [select, locked to thread]:
github.com/hajimehoshi/ebiten/v2/internal/thread.(*OSThread).Loop(0xc000322a80)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/thread/thread.go:47 +0x87
github.com/hajimehoshi/ebiten/v2/internal/uidriver/glfw.(*UserInterface).runOnAnotherThreadFromMainThread(0x645700, 0xc000066940)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/uidriver/glfw/run_notsinglethread.go:80 +0x23b
github.com/hajimehoshi/ebiten/v2/internal/uidriver/glfw.(*UserInterface).registerWindowSetSizeCallback.func1(0xc000034000, 0x245, 0x1f5)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/uidriver/glfw/ui.go:748 +0xc5
github.com/hajimehoshi/ebiten/v2/internal/glfw.ToSizeCallback.func1(0xc000217810, 0x2, 0x3)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/glfw/callback_windows.go:68 +0x53
syscall.Syscall(0x57aaeabf, 0x3, 0x1ba0a908440, 0x245, 0x1f6)
        /usr/local/go/src/runtime/syscall_windows.go:483 +0xf4
golang.org/x/sys/windows.(*Proc).Call(0x0, {0xc0000f5c80, 0xc000057500, 0x18})
        /home/tom/go/pkg/mod/golang.org/x/sys@v0.0.0-20210917161153-d61c044b1678/windows/dll_windows.go:174 +0x185
golang.org/x/sys/windows.(*LazyProc).Call(0xc0001419e0, {0xc0000f5c80, 0x3, 0x3})
        /home/tom/go/pkg/mod/golang.org/x/sys@v0.0.0-20210917161153-d61c044b1678/windows/dll_windows.go:348 +0x50
github.com/hajimehoshi/ebiten/v2/internal/glfw.(*dll).call(0xc000040640, {0x41a78e, 0xc000217e14}, {0xc0000f5c80, 0x3, 0x3})
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/glfw/load_windows.go:37 +0x173
github.com/hajimehoshi/ebiten/v2/internal/glfw.(*Window).SetKeepAspectRatio(0xc0005880a0, 0x0)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/glfw/glfw_windows.go:269 +0x95
github.com/hajimehoshi/ebiten/v2/internal/uidriver/glfw.(*window).SetKeepAspectRatio.func1()
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/uidriver/glfw/window.go:236 +0x28
github.com/hajimehoshi/ebiten/v2/internal/thread.(*OSThread).Loop.func1(0xc000217e20, 0xc000217e10)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/thread/thread.go:54 +0x53
github.com/hajimehoshi/ebiten/v2/internal/thread.(*OSThread).Loop(0xc00053c000)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/thread/thread.go:55 +0x2e
github.com/hajimehoshi/ebiten/v2/internal/uidriver/glfw.(*UserInterface).Run(0x645700, {0x456e30, 0x64a060})
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/uidriver/glfw/run_notsinglethread.go:54 +0x1df
github.com/hajimehoshi/ebiten/v2.RunGame({0x456768, 0x69f790})
        /home/tom/go/src/github.com/hajimehoshi/ebiten/run.go:163 +0xf9
main.main()
        /home/tom/go/src/github.com/hajimehoshi/ebiten/examples/blur/main.go:99 +0x145

goroutine 34 [chan receive]:
github.com/hajimehoshi/ebiten/v2/internal/thread.(*OSThread).Call(0xc00053c000, 0x2b8620)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/thread/thread.go:74 +0x3e
github.com/hajimehoshi/ebiten/v2/internal/uidriver/glfw.(*window).SetKeepAspectRatio(0x647c80, 0x1)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/uidriver/glfw/window.go:235 +0x8c
github.com/hajimehoshi/ebiten/v2.SetWindowKeepAspectRatio(0x13)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/window.go:226 +0x3d
main.(*Game).Update(0xc000502480)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/examples/blur/main.go:43 +0x33
github.com/hajimehoshi/ebiten/v2.(*imageDumper).update(0xc00058c640)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/imagedumper_desktop.go:111 +0x52
github.com/hajimehoshi/ebiten/v2.(*imageDumperGame).Update(0x64a060)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/run.go:118 +0x7d
github.com/hajimehoshi/ebiten/v2.(*uiContext).updateFrameImpl(0x64a060, 0x1)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/uicontext.go:188 +0xc9
github.com/hajimehoshi/ebiten/v2.(*uiContext).updateFrame(0x64a060, 0x4018e0)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/uicontext.go:160 +0x86
github.com/hajimehoshi/ebiten/v2.(*uiContext).UpdateFrame(0xc00053c000)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/uicontext.go:139 +0x32
github.com/hajimehoshi/ebiten/v2/internal/uidriver/glfw.(*UserInterface).loop(0x645700)
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/uidriver/glfw/ui.go:1060 +0x1d0
github.com/hajimehoshi/ebiten/v2/internal/uidriver/glfw.(*UserInterface).Run.func1()
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/uidriver/glfw/run_notsinglethread.go:47 +0x13d
created by github.com/hajimehoshi/ebiten/v2/internal/uidriver/glfw.(*UserInterface).Run
        /home/tom/go/src/github.com/hajimehoshi/ebiten/internal/uidriver/glfw/run_notsinglethread.go:34 +0x1bd
```

**Comment by tomlister** [comment]
I triggered this by spamming buttons I bound to call the method

**Comment by hajimehoshi** [comment]
Does this panic happens with your fix, or even without?

**Comment by tomlister** [comment]
I'm not certain. I introduce some locking/unlocking, but it's the same as any other initwindow methods.

**Comment by hajimehoshi** [comment]
Any updates?

**Comment by hajimehoshi** [comment]
Could you rebase this? Unfortunately, many files have been moved. e.g., `internal/uidriver/glfw/ui.go` -> `internal/ui/ui_glfw.go`

**Comment by tomlister** [comment]
I am so confused

**Comment by hajimehoshi** [comment]
Creating another PR is also fine. I can help if you need.

**Pull request closed without merging**